application_manager/plugins/spark_mesos/plugin.py

```diff
@@ -0,0 +1,250 @@
+                       (api.one_username, api.one_password, api.one_url)
+        stdin, stdout, stderr = conn.exec_command(list_vms_one)
+
+        list_response = stdout.read()
+        vms_ips, master = self._get_executors_ip(conn)
+        vms_ids = self._extract_vms_ids(list_response)
+
+        executors_vms_ids = []
+        for ip in vms_ips:
+            for id in vms_ids:
+                vm_info_one = 'onevm show %s --user %s --password %s \
+                              --endpoint %s' % (id, api.one_username,
+                                                api.one_password, api.one_url)
+
+                stdin, stdout, stderr = conn.exec_command(vm_info_one)
+                if ip in stdout.read():
+                    executors_vms_ids.append(id)
+                    break
+
+        plugin_log.log("%s | Executors IDs: %s"
+                        % (time.strftime("%H:%M:%S"), executors_vms_ids))
+
+        # Set up the initial configuration of cpu cap
+        scaler.setup_environment(api.controller_url, executors_vms_ids,
+                                 starting_cap, data)
+
+        info_plugin = {"spark_submisson_url": master,
+                       "expected_time": expected_time,
+                       "number_of_jobs": number_of_jobs}
+
+        plugin_log.log("%s | Starting monitor" % (time.strftime("%H:%M:%S")))
+        monitor.start_monitor(api.monitor_url, self.app_id,
+                              'spark-mesos', info_plugin, 2)
+
+        plugin_log.log("%s | Starting scaler" % (time.strftime("%H:%M:%S")))
+        scaler.start_scaler(api.controller_url,
+                            self.app_id,
+                            executors_vms_ids,
+                            data)
+
+        # This command locks the plugin execution until the execution be done
+        print o.read()
+
+        plugin_log.log("%s | Stopping monitor" % (time.strftime("%H:%M:%S")))
+        monitor.stop_monitor(api.monitor_url, self.app_id)
+
+        plugin_log.log("%s | Stopping scaler" % (time.strftime("%H:%M:%S")))
+        scaler.stop_scaler(api.controller_url, self.app_id)
+
+        plugin_log.log("%s | Remove binaries" % (time.strftime("%H:%M:%S")))
+        conn.exec_command('rm -rf ~/exec_bin.*')
+        return None, None
+
+    def _get_ssh_connection(self, ip, username=None,
+                            password=None, key_path=None):
+        # Preparing SSH connection
+        conn = paramiko.SSHClient()
+        conn.set_missing_host_key_policy(paramiko.AutoAddPolicy())
+
+        # Checking if the connection will be established using
+        # keys or passwords
+        if key_path != "" and key_path is not None:
+            keypair = paramiko.RSAKey.from_private_key_file(key_path)
+            conn.connect(hostname=ip, username=username, pkey=keypair)
+        else:
+            conn.connect(hostname=ip, username=username, password=password)
+
+        return conn
+
+    def _get_executors_ip(self, conn):
```

Comment:
I think this method should be moved to "utils/mesos.py"

Reply:
I agree.